packages/tldraw/src/lib/shapes/arrow/ArrowShapeUtil.tsx

```diff
@@ -935,40 +927,26 @@ const ArrowSvg = track(function ArrowSvg({
 	const maskStartArrowhead = !(info.start.arrowhead === 'none' || info.start.arrowhead === 'arrow')
 	const maskEndArrowhead = !(info.end.arrowhead === 'none' || info.end.arrowhead === 'arrow')
 
-	// NOTE: I know right setting `changeIndex` hacky-as right! But we need this because otherwise safari loses
-	// the mask, see <https://linear.app/tldraw/issue/TLD-1500/changing-arrow-color-makes-line-pass-through-text>
-	const maskId = sanitizeId(shape.id + '_clip' + (tlenv.isSafari ? `_${changeIndex}` : ''))
```

Comment:
Looks like this does not apply to `clipPaths`, which is a nice win on top of fixing perf.